internal/webhook/bmc/v1beta1/virtualmachinebmc_webhook.go

```diff
@@ -73,54 +79,121 @@ func (d *VirtualMachineBMCCustomDefaulter) Default(_ context.Context, obj runtim
+		return nil, err
+	}
+
+	if err := v.validateAuthSecretRef(ctx, virtualmachinebmc); err != nil {
+		return nil, err
+	}
 
 	return nil, nil
 }
 
 // ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type VirtualMachineBMC.
-func (v *VirtualMachineBMCCustomValidator) ValidateUpdate(_ context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
-	virtualmachinebmc, ok := newObj.(*virtualmachinev1alpha1.VirtualMachineBMC)
+func (v *VirtualMachineBMCCustomValidator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
+	virtualmachinebmc, ok := newObj.(*bmcv1beta1.VirtualMachineBMC)
 	if !ok {
 		return nil, fmt.Errorf("expected a VirtualMachineBMC object for the newObj but got %T", newObj)
 	}
 	virtualmachinebmclog.Info("Validation for VirtualMachineBMC upon update", "name", virtualmachinebmc.GetName())
 
-	// TODO(user): fill in your validation logic upon object update.
+	if err := v.validateVirtualMachineRef(ctx, virtualmachinebmc); err != nil {
+		return nil, err
+	}
+
+	if err := v.validateAuthSecretRef(ctx, virtualmachinebmc); err != nil {
+		return nil, err
+	}
 
 	return nil, nil
 }
 
 // ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type VirtualMachineBMC.
 func (v *VirtualMachineBMCCustomValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
-	virtualmachinebmc, ok := obj.(*virtualmachinev1alpha1.VirtualMachineBMC)
+	virtualmachinebmc, ok := obj.(*bmcv1beta1.VirtualMachineBMC)
 	if !ok {
 		return nil, fmt.Errorf("expected a VirtualMachineBMC object but got %T", obj)
 	}
 	virtualmachinebmclog.Info("Validation for VirtualMachineBMC upon deletion", "name", virtualmachinebmc.GetName())
 
-	// TODO(user): fill in your validation logic upon object deletion.
-
+	// No validation needed for deletion
 	return nil, nil
 }
+
+// validateVirtualMachineRef checks if the referenced VirtualMachine exists in the same namespace.
+func (v *VirtualMachineBMCCustomValidator) validateVirtualMachineRef(ctx context.Context, virtualmachinebmc *bmcv1beta1.VirtualMachineBMC) error {
+	if virtualmachinebmc.Spec.VirtualMachineRef == nil {
+		return fmt.Errorf("spec.virtualMachineRef is required")
+	}
+
+	if virtualmachinebmc.Spec.VirtualMachineRef.Name == "" {
+		return fmt.Errorf("spec.virtualMachineRef.name cannot be empty")
+	}
+
+	vmKey := client.ObjectKey{
+		Namespace: virtualmachinebmc.Namespace,
+		Name:      virtualmachinebmc.Spec.VirtualMachineRef.Name,
+	}
+
+	var vm kubevirtv1.VirtualMachine
+	if err := v.Client.Get(ctx, vmKey, &vm); err != nil {
+		if apierrors.IsNotFound(err) {
+			return fmt.Errorf("referenced VirtualMachine %q not found in namespace %q",
+				virtualmachinebmc.Spec.VirtualMachineRef.Name,
+				virtualmachinebmc.Namespace)
+		}
+		return fmt.Errorf("error checking VirtualMachine existence: %w", err)
+	}
+
+	return nil
+}
```

Comment:
The same check exists in the controller code. Is it possible to make it a util function leverage by both the controller and the validating webhook?